src/language-js/print/misc.js

```diff
@@ -55,6 +55,10 @@ function printTypeAnnotation(path, options, print) {
 
   const parentNode = path.getParentNode();
 
+  if (parentNode.type === "ArrayPattern" && options.parser === "babel-ts") {
+    return [" as ", print("typeAnnotation")];
+  }
```

Comment:
Ideally we should fix this in babel-parser side, but this seems to be work well as temporary fix.

Reply:
Has this been reported?

Reply:
done https://github.com/babel/babel/issues/14498